src/hypothesis/strategies.py

```diff
@@ -1205,6 +1223,25 @@ def dm_func(denom):
         lambda f: f.limit_denominator(max_denominator))
 
 
+def _as_finite_decimal(value, name, allow_infinity):
+    """Convert decimal bounds to decimals, carefully."""
+    assert name in ('min_value', 'max_value')
+    if value is None:
+        return None
+    if not isinstance(value, Decimal):
+        with localcontext(Context()):  # ensure that default traps are enabled
```

Comment:
Do you have/want a test that this part does what it's supposed to?

Reply:
I do now!

(and I had to add `ArithmeticError` to the validation in bounds checks, so good catch :smile:)